components/tidb_query_expr/src/impl_time.rs

```diff
@@ -218,6 +219,94 @@ pub fn make_date(ctx: &mut EvalContext, year: &Int, day: &Int) -> Result<Option<
     Ok(Some(ret))
 }
 
+pub fn make_time(
```

Comment:
You can reuse [`Duration::new_from_parts` ](https://tikv.github.io/doc/tidb_query_datatype/codec/mysql/duration/struct.Duration.html#method.new_from_parts)

Reply:
This can be used in the last line to replace the parse().

Reply:
> This can be used in the last line to replace the parse().

I see

Reply:
`MAX_HOUR_PART` and `MAX_MININTE_PART` are also useful. (https://tikv.github.io/doc/tidb_query_datatype/codec/mysql/duration/index.html)

Reply:
> You can reuse [`Duration::new_from_parts` ](https://tikv.github.io/doc/tidb_query_datatype/codec/mysql/duration/struct.Duration.html#method.new_from_parts)

I see ,and we should handle error without ctx? or by self?

Reply:
If I'm right(probably not), this function is infallible? Since all overflows are caught.

Reply:
I mean, you may keep the overflow checked, and call new_from_parts().unwrap().